### PR TITLE
Use `random_state` parameter in KMeans clustering for reproducibility

### LINKED ISSUE #812: Allow for `random_state` parameter passing to neighborhood analysis `KMeans` functions
**Is your feature request related to a problem? Please describe.**

Because `KMeans` is a non-deterministic algorithm, it will produce different results each time it's run. For reproducibility purposes, it will be helpful to define a seed parameter to pass throughout neighborhood analysis.

**Describe the solution you'd like**

The parameter is `random_state` in `KMeans`. We can alias that with `seed` in `generate_cluster_matrix_results` for consistency purposes.

This will need to be addressed for the following functions:

- `compute_cluster_metrics_inertia`
- `compute_cluster_metrics_silhouette`
- `generate_cluster_matrix_results`

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Closes #812. Adds a random seed to KMeans clustering in neighborhood analysis to ensure reproducible results.

**How did you implement your changes**

Pass a `seed` parameter to the cluster metrics and assignment functions of neighborhood analysis, which is analogous with `random_state` in `sklearn.cluster.KMeans` and `pd.DataFrame.sample`.

### CODE REVIEW

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Did you run through the whole notebook with the same data twice to make sure you can replicate the same output?

**Review by cliu72 — COMMENTED** [review]
Looks good to me
